gix-traverse/Cargo.toml

```diff
@@ -20,5 +20,7 @@ gix-date = { version = "^0.8.5", path = "../gix-date" }
 gix-hashtable = { version = "^0.5.2", path = "../gix-hashtable" }
 gix-revwalk = { version = "^0.13.0", path = "../gix-revwalk" }
 gix-commitgraph = { version = "^0.24.2", path = "../gix-commitgraph" }
+gix-revision = { version = "^0.27.0", path = "../gix-revision" }
 smallvec = "1.10.0"
 thiserror = "1.0.32"
+flagset = "0.4.4"
```

Comment:
This should most certainly be `bitflags` as this crate is already used in the dependency tree.

Reply:
Ah, yeah that would make more sense. I wasn't aware bitflags was in use so when I searched around it seemed like `flagset` was the new shiny thing to use. WIll convert to `bitflags`.

On that topic... The whole construct of using a bitfield for this is taken straight from Git, where there are many many more fields and the topological sort just adds a few. This thing *could* just be a `struct` with a couple of booleans if we wanted to. Another idea would be to use some sort of typestate thing where a commit changes state from been `Seen` to being `Explored`. which would very much make sense and mimic how the different "walks" happen.

Reply:
Thanks for elaborating! I took another look at `flagset` and loved that it comes with only 410 lines of code. Otherwise, it can probably cover everything that `bitflags` is currently used for.

That's probably something for another PR, but I certainly wouldn't mind switching. If you consider doing that please note that `gix-index` is the place to look first, as it has the most elaborate use of `bitflags` that should be covered.